### PR TITLE
NIFI-10702 Clear server list on connection error in SMB processors

### PR DESCRIPTION
**Pull request by kulikg**
# Summary

[NIFI-10702](https://issues.apache.org/jira/browse/NIFI-10702)

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 8
  - [x] JDK 11
  - [x ] JDK 17

### Licensing

- [x] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [x] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [x] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-smb-bundle/nifi-smb-smbj-client/src/main/java/org/apache/nifi/services/smb/SmbjClientProviderService.java [hunk 1]
```diff
@@ -129,6 +129,7 @@ public SmbClientService getClient() throws IOException {
             getLogger().debug("Closing stale connection and trying to create a new one for share " + getServiceLocation());
 
             closeConnection(connection);
+            unregisterHost();
```
Comment (tpalfy) [inline nifi-nar-bundles/nifi-smb-bundle/nifi-smb-smbj-client/src/main/java/org/apache/nifi/services/smb/SmbjClientProviderService.java:132]: Unfortunately
```java
Connection connection = smbClient.connect(hostname, port);
```
can also fail for the same reason this change is trying to fix.

To my understanding this is because how the connection is cached by the SMBClient.
Our current approach increases the number of leases on the cached connection every time we ask for one. And those leases are never "returned" by us. So the number of leases increases indefinitely and can almost never reach zero. Except when we have asked for a connection _only once_. If the connection is closed after this first occurrence the number of leases reaches zero and it is discarded.

However the server hostname is already stored so if that changes an exception comes when we ask the client to create a new connection.

Long story short, the
```java
smbClient.connect(hostname, port);
```
call should be in a try-catch block as well.

**Review by tpalfy — COMMENTED** [review]
Since bcprov-jdk15on is excluded as a transitive dependency, we need to add
```xml
            <dependency>
                <groupId>org.bouncycastle</groupId>
                <artifactId>bcprov-jdk18on</artifactId>
            </dependency>
```
in both nifi-nar-bundles/nifi-smb-bundle/nifi-smb-processors/pom.xml and nifi-nar-bundles/nifi-smb-bundle/nifi-smb-smbj-client/pom.xml.

We also need to remove the same entry from the dependencyManagement in nifi-nar-bundles/nifi-smb-bundle/pom.xml.

**Review by turcsanyip — APPROVED** [review]
+1 LGTM

**Comment by tpalfy** [comment]
LGTM+1
